NuRadioMC/simulation/simulation.py

```diff
@@ -558,53 +572,38 @@ def run(self):
-                                       n_reflections=self._n_reflections)
-
+                        self._raytracer.set_start_and_end_point(x1, x2)
                         if(pre_simulated and ray_tracing_performed and not self._cfg['speedup']['redo_raytracing']):  # check if raytracing was already performed
                             sg_pre = self._fin_stations["station_{:d}".format(self._station_id)]
-                            temp_reflection = None
-                            temp_reflection_case = None
-                            if('ray_tracing_reflection' in sg_pre):  # for backward compatibility: Check if reflection layer information exists in data file
-                                temp_reflection = sg_pre['ray_tracing_reflection'][self._shower_index][channel_id]
-                                temp_reflection_case = sg_pre['ray_tracing_reflection_case'][self._shower_index][channel_id]
-                            r.set_solution(sg_pre['ray_tracing_C0'][self._shower_index][channel_id],
-                                           sg_pre['ray_tracing_C1'][self._shower_index][channel_id],
-                                           sg_pre['ray_tracing_solution_type'][self._shower_index][channel_id],
-                                           temp_reflection, temp_reflection_case)
+                            self._raytracer.set_solution(sg_pre, self._shower_index, channel_id)
                         else:
-                            r.find_solutions()
+                            self._raytracer.find_solutions()
 
-                        if(not r.has_solution()):
+                        if(not self._raytracer.has_solution()):
                             logger.debug("event {} and station {}, channel {} does not have any ray tracing solution ({} to {})".format(
                                 self._event_group_id, self._station_id, channel_id, x1, x2))
                             continue
                         delta_Cs = []
                         viewing_angles = []
                         # loop through all ray tracing solution
-                        for iS in range(r.get_number_of_solutions()):
-                            sg['ray_tracing_C0'][iSh, channel_id, iS] = r.get_results()[iS]['C0']
-                            sg['ray_tracing_C1'][iSh, channel_id, iS] = r.get_results()[iS]['C1']
-                            sg['ray_tracing_reflection'][iSh, channel_id, iS] = r.get_results()[iS]['reflection']
-                            sg['ray_tracing_reflection_case'][iSh, channel_id, iS] = r.get_results()[iS]['reflection_case']
-                            sg['ray_tracing_solution_type'][iSh, channel_id, iS] = r.get_solution_type(iS)
-                            self._launch_vector = r.get_launch_vector(iS)
+                        for iS in range(self._raytracer.get_number_of_solutions()):
+                            self._raytracer.write_raytracing_output(sg, iSh, channel_id, iS)
```

Comment:
this interface should be different too. The ray tracer should return a dict with the parameters that define the solution. simulation.py should take case how it is written into the hdf5 file

Reply:
Done